config-linux.md

```diff
@@ -642,6 +642,7 @@ The values MUST be absolute paths in the [container namespace](glossary.md#conta
 ## <a name="configLinuxMountLabel" />Mount Label
 
 **`mountLabel`** (string, OPTIONAL) will set the Selinux context for the mounts in the container.
+    The value MUST NOT be an empty string.
```

Comment:
> So i dont think opencontainers/runtime-tools#386 that you referred to has anything to do with empty string clarification here.

The connection is [the line I linked in the initial comment][1].

    if v.spec.Linux.MountLabel != "" {
        …validate the value
    }

That only makes sense if we have some reason to know that either:

a. the empty string is valid in its own right (in which case we'd probably want to patch the Go type to use `*string` and patch the validation to accept the empty string), or
b. the empty string is invalid, so we can use something like opencontainers/runtime-tools#197 if we want to catch those invalid ,`"mountLabel": ""` configs, and everybody else can do whatever undefined thing they want them (including treating them like configs that don't set `mountLabel` at all, which is what the `string` Go type does).

This PR goes with (b), but I'm fine going with (a) if folks feel like that's more appropriate.

> …i think it doesnt make sense for the spec to chase how selinux validates the options…

I think *this* part of validation is #844.

> IIRC we just remove null / pointers in many fields because it's replaceable in many cases with the usages of default value which is simpler to use ( in this case an empty string ).

There's a distinction between “the default is {whatever}”, where the runtime still *does something* with that default (e.g. [here][2]), and “the default is to not do anything”, where you might as well leave the property unset.  We have lots of cases where leaving the property unset means “do nothing”, and I don't see the point of defining (and requiring support for) synonyms like setting an explicit `null` or empty-string value.

> I'm not sure if this is true, but even if it is, clarification on the default value ( empty string ) seems like a more reasonable option for me than enforcing that it always needs to have an non-empty value.

That's more work for runtimes (now you have to handle both unset and empty-string values), although it's not much more work in Go (where using a `string` type and the default deserializer collapse those two cases for you).  And it's more work for compliance testing, which now needs to check that the runtime does the right thing for both the unset and empty-string cases.  And it's an unnecessary decision for config authors, who need to figure out which of the two ways to say “don't do anything” they'd like to use.  What's the upside?

[1]: https://github.com/opencontainers/runtime-tools/pull/386/commits/b92ab4284b90ab07c529ac69a0f647c15a706f50#diff-8351286f57eb81e245c9c99c07f3b34fR523
[2]: https://github.com/opencontainers/runtime-spec/blame/4d51b5994148738a0c9adb2cd3534bf0be4c1322/config-linux.md#L221